extensions/amp-selector/1.0/amp-selector.js

```diff
@@ -249,12 +250,42 @@ function OptionShim({
 function SelectorShim({
   shimDomElement,
   children,
+  form,
   multiple,
+  name,
   disabled,
   onKeyDown,
   role = 'listbox',
   tabIndex,
+  value,
 }) {
+  const input = useRef(
+    createElementWithAttributes(shimDomElement.ownerDocument, 'input', {
```

Comment:
The `createElementWithAttributes` will be called each time here, which is rather expensive, since only the first value is used. Unfortunately, it might have to be an ugly-ish:

```
const input = useRef();
if (!input.current) {
  input.current = createElementWithAttributes(...);
}
```

Reply:
Would we want to `useMemo` instead?

Reply:
`useMemo` doesn't guarantee 100% stability. It's more like cache. Though I've never seen it reset. It's definitely possible to use `useState` of course. E.g.

```
const [input] = useState(() => createElementwithAttrs(...));
```

I guess it's more of a matter of style between a ref and a state in this case.

Reply:
I'll keep with the ref for now.